File: array/209.长度最小的子数组.go
/*
 * @lc app=leetcode.cn id=209 lang=golang
 *
 * [209] 长度最小的子数组
 */

// @lc code=start
func minSubArrayLen(target int, nums []int) int {
	l := len(nums)
	left := 0
	res := math.MaxInt
	sum := 0

	for right := 0; right < l; right++ {
		sum += nums[right]
		for sum >= target {
			temp := right - left + 1
			if temp < res {
				res = temp
			}
			sum -= nums[left]
			left++
		}
	}

	if res == math.MaxInt {
		res = 0
	}

	return res
}

// @lc code=end

File: array/209.长度最小的子数组.py
#
# @lc app=leetcode.cn id=209 lang=python3
#
# [209] 长度最小的子数组
#

# @lc code=start
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        right = 0
        n = len(nums) - 1
        sum = 0
        res = sys.maxsize   

        while right <= n:
            sum += nums[right]
            while sum >= target:
                temp = right - left + 1
                res = min(res, temp) 
                sum -= nums[left]
                left += 1
            right += 1
        return res if res != sys.maxsize else 0
# @lc code=end

